snuba/datasets/entities/assign_reason.py

```diff
@@ -0,0 +1,65 @@
+from typing import Any, Mapping, Optional, Sequence
+
+
+def assign_reason_category(
+    data: Sequence[Mapping[str, Any]],
+    expected_data: Sequence[Mapping[str, Any]],
+    referrer: str,
+) -> str:
+    """
+    Attempt to categorize the reason for the discrepancy.
+    Mostly based on the type of query we expect for each referrer.
+    """
+
+    try:
+        if referrer.startswith("tsdb-modelid:"):
+            agg = check_aggregate(data, expected_data, "aggregate")
+            if agg is not None:
+                return agg
+
+        if referrer == "tagstore.get_groups_user_counts":
+            agg = check_aggregate(data, expected_data, "count")
+            if agg is not None:
+                return agg
+
+        if referrer == "api.serializer.projects.get_stats":
+            agg = check_aggregate(data, expected_data, "count")
+            if agg is not None:
+                return agg
+
+        if referrer == "tagstore.__get_tag_key_and_top_values":
+            agg = check_aggregate(data, expected_data, "count")
+            if agg is not None:
+                return agg
+
+        return "UNKNOWN"
+
+    except Exception:
+        return "UNKNOWN"
+
+
+def check_aggregate(
+    data: Sequence[Mapping[str, Any]],
+    expected_data: Sequence[Mapping[str, Any]],
+    aggregate_key: str,
+) -> Optional[str]:
+    """
+    Returns AGGREGATE_TOO_HIGH or AGGREGATE_TOO_LOW if the result is the same
+    except for the aggregate value. Could be an indication of differences in how
+    parts are merged between the two tables, timing differences or something else.
+    """
+    if len(data) == len(expected_data):
+        for idx in range(len(data)):
```

Comment:
All queries have an order by, but it is still possible for results to be out of order. For example order by on event_id will yield completely different results from both tables since string and UUID have completely different ordering. Catching incorrectly ordered but otherwise same results is something I'd like to add in this function in future.